magic-nix-cache/src/github.rs

```diff
@@ -0,0 +1,52 @@
+use serde::{Deserialize, Serialize};
+
+const GITHUB_ACTOR_TYPE_USER: &str = "User";
+const GITHUB_ACTOR_TYPE_ORGANIZATION: &str = "Organization";
+
+#[derive(Serialize, Deserialize)]
+pub struct WorkflowData {
+    event: WorkflowDataEvent,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct WorkflowDataEvent {
+    repository: WorkflowDataEventRepo,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct WorkflowDataEventRepo {
+    owner: WorkflowDataEventRepoOwner,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct WorkflowDataEventRepoOwner {
+    login: String,
+    #[serde(rename = "type")]
+    kind: String,
+}
+
+pub(crate) fn get_actions_event_data() -> color_eyre::Result<WorkflowData> {
+    let github_context = std::env::var("GITHUB_CONTEXT")?;
+    let workflow_data: WorkflowData = serde_json::from_str::<WorkflowData>(&github_context)?;
+
+    Ok(workflow_data)
+}
+
+pub(crate) fn print_unauthenticated_error() {
+    let mut msg = "::error title=FlakeHub registration required.::Unable to authenticate to FlakeHub. Individuals must register at FlakeHub.com; Organizations must create an organization at FlakeHub.com.".to_string();
+    if let Ok(workflow_data) = get_actions_event_data() {
```

Comment:
I think we may not want to print the message if we get None event data, in case people are pinning the action source but not the binary.

Reply:
nevermind, this will be correct in that case anyway!